src/util/throttle.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+
+/**
+ * Throttle the given function to run at most every `period` milliseconds.
+ */
+module.exports = function(unthrottledFunction: () => void, period: number): () => void {
+
+    // The next time (unix epoch) that the function is allowed to execute
+    let nextTime = 0;
+
+    // `true` if there is a pending "setTimeout" operation that'll invoke the
+    // function at a later time. `false` if there is not.
+    let pending = false;
+
+    const throttledFunction = () => {
+        const time = Date.now();
+
+        if (nextTime <= time && !pending) {
+            nextTime = time + period;
+            unthrottledFunction();
+        } else if (!pending) {
+            pending = true;
+            // eslint-disable-next-line no-use-before-define
```

Comment:
The [ESLint docs](https://eslint.org/docs/rules/no-use-before-define) on the rule state that:

> In ES6, block-level bindings (let and const) introduce a “temporal dead zone” where a ReferenceError will be thrown with any attempt to access the variable before its declaration.

So if this code weren't transpiled (which may happen in future) it would through an error in theory. So perhaps let's rewrite to either use `function` declaration instead of arrows, or declare one of the vars at the top with `let`?

Reply:
Whoa! I've been out in the woods too long. Fixed in 20f3fe0.